src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let ACTION_REQUIRED = '';
+  const stateCopy = { ...state };
+  const ARRAY = [Object.assign({}, stateCopy)];
+
+  for (const action of actions) {
+    ACTION_REQUIRED = action.type;
+    let elementArray = Object.assign({}, ARRAY[ARRAY.length - 1]);
```

Comment:
let elementArray = Object.assign({}, ARRAY[ARRAY.length - 1]); can be simplified to let elementArray = { ...ARRAY[ARRAY.length - 1] };

Reply:
Fixed.